.github/workflows/nestbuildmatrix.yml

```diff
@@ -2,32 +2,444 @@
 # detailed syntax defined in
 # https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions
 #
-name: "NEST matrix jobs"
-env:
-  CXX_FLAGS: "-pedantic -Wextra -Woverloaded-virtual -Wno-unknown-pragmas"
-  PYTHONPATH: ${{ github.workspace }}/build/python
-on: [push, pull_request]
+name: "Test suite"
 
+on: [push, pull_request]
 
 jobs:
+  clang-format:
+    runs-on: "ubuntu-20.04"
+    env:
+      CLANG_REQUIRE_VERSION: 13
+      CLANG_FORMAT_FILE: ".clang-format"
+    steps:
+      - name: "Checkout repository content"
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: "Install Linux system dependencies"
+        run: |
+          python -m pip install --force-reinstall clang-format==${{ env.CLANG_REQUIRE_VERSION }}
+          clang-format --version
+
+      - name: "Run clang-format..."
+        run: |
+          files=$(find . \( -iname "*.h" -o -iname "*.c" -o -iname "*.cc" -o -iname "*.cpp" \) -not -path "./thirdparty/*")
+          diff -u <(cat ${files}) <(clang-format ${files})
+
+  cppcheck:
+    runs-on: "ubuntu-20.04"
+    steps:
+      - name: "Checkout repository content"
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: "Install Linux system dependencies"
+        run: |
+          sudo apt-get update
+          sudo apt-get install cppcheck
+
+      - name: "Run cppcheck..."
+        run: |
+          cppcheck --enable=all --language=c++ --std=c++11 -i sli/ --suppressions-list=.cppcheck_suppressions ./
+
+  rstcheck:
+    runs-on: "ubuntu-20.04"
+    steps:
+      - name: "Checkout repository content"
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: "Set up Python 3.x"
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.9
+
+      - name: "Install dependencies"
+        run: |
+          pip install rstcheck
+
+      - name: "Run rstcheck..."
+        run: |
+          rstcheck -r doc/
+
+  vale-action:
+    runs-on: "ubuntu-20.04"
+    steps:
+      - name: "Checkout repository content"
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: "Vale documentation linting"
+        uses: errata-ai/vale-action@v1.5.0
+        # 1.0.1
+        #with:
+        #  config: https://raw.githubusercontent.com/terhorstd/nest-simulator/separated-checks-vale/.vale.ini
+        env:
+          # Required, set by GitHub actions automatically:
+          # https://docs.github.com/en/actions/security-guides/automatic-token-authentication#about-the-github_token-secret
```

Comment:
```suggestion
```
Probably all lines in this repository are "required", otherwise they would not be here...